src/models/flinkUDF.test.ts

```diff
@@ -0,0 +1,100 @@
+import assert from "assert";
+import { describe, it } from "mocha";
+import { createFlinkUDF } from "../../tests/unit/testResources/flinkUDF";
+import { FlinkUdfParameter, FlinkUdfTreeItem, createFlinkUdfToolTip } from "./flinkUDF";
+import { CCLOUD_CONNECTION_ID } from "../constants";
+import { ConnectionType } from "../clients/sidecar";
+
+describe("FlinkUdfParameter", () => {
+  describe("formatSqlType", () => {
+    it("should remove max int size", () => {
+      const result = FlinkUdfParameter.formatSqlType("VARCHAR(2147483647)");
+      assert.strictEqual(result, "VARCHAR");
+    });
+
+    it("should preserve small sizes", () => {
+      const result = FlinkUdfParameter.formatSqlType("VARCHAR(100)");
+      assert.strictEqual(result, "VARCHAR(100)");
+    });
+
+    it("should remove backticks", () => {
+      const result = FlinkUdfParameter.formatSqlType("ROW<`field` VARCHAR>");
+      assert.strictEqual(result, "ROW<field VARCHAR>");
+    });
+
+    it("should handle complex types with max int and backticks", () => {
+      const result = FlinkUdfParameter.formatSqlType("ROW<`name` VARCHAR(2147483647), `age` INT>");
+      assert.strictEqual(result, "ROW<name VARCHAR, age INT>");
+    });
+  });
+});
+
+describe("FlinkUdf", () => {
+  it("should return correct details", () => {
+    const udf = createFlinkUDF("testFunc");
```

Comment:
could add `description` to the `opts` definition of `createFlinkUDF` real easy, then no trailing assignment here. That function and its `opts` is expressly to make tests easier, and it tests need more, then go ahead and make it happen!